src/platform_impl/linux/wayland/seat/touch/mod.rs

```diff
@@ -42,18 +42,26 @@ impl TouchHandler for WinitState {
         let location = LogicalPosition::<f64>::from(position);
         seat_state.touch_map.insert(id, TouchPoint { surface, location });
 
+        let device_id = crate::event::DeviceId(crate::platform_impl::DeviceId::Wayland(DeviceId));
+        let position = location.to_physical(scale_factor);
+        let finger_id =
+            crate::event::FingerId(crate::platform_impl::FingerId::Wayland(FingerId(id)));
+
         self.events_sink.push_window_event(
-            WindowEvent::Touch(Touch {
-                device_id: crate::event::DeviceId(crate::platform_impl::DeviceId::Wayland(
-                    DeviceId,
-                )),
-                phase: TouchPhase::Started,
-                location: location.to_physical(scale_factor),
-                force: None,
-                finger_id: crate::event::FingerId(crate::platform_impl::FingerId::Wayland(
-                    FingerId(id),
-                )),
-            }),
+            WindowEvent::PointerEntered {
+                device_id,
+                position,
+                kind: PointerKind::Touch(finger_id),
+            },
+            window_id,
+        );
```

Comment:
Weren't we discussing that this should be emitted only for the first any finger touching the surface, so it'll look more like a pointer focus stuff?

Reply:
Don't remember :cold_sweat:.

I'm not sure what the right move is here, but I would argue that this makes more sense because it is simply more accurate. Unsure how not emitting `PointerEntered` would make it look more like pointer focus.

Currently I'm in favor of leaving it as-is because its accurate and I'm not sure what the advantage would be of changing it.

Reply:
In the current state of things there's no need for `PointerButton` for touch at all, since you can not touch without enter/leave meaning that enter/leave already model the touch down/up.

It's just events are redundant, since they end up doing the exact same thing, indicating touch down/up...

From what I can tell, maybe on some platforms you want to react when you get pointer focus/leave focus, but with touch you'll run this state change for each finger, which doesn't look _right_. Imagine decorations lightening, etc.

Reply:
> In the current state of things there's no need for `PointerButton` for touch at all, since you can not touch without enter/leave meaning that enter/leave already model the touch down/up.
> 
> It's just events are redundant, since they end up doing the exact same thing, indicating touch down/up...

Hm ... agreed!
Well, lets hash it out at the meeting today.
> From what I can tell, maybe on some platforms you want to react when you get pointer focus/leave focus, but with touch you'll run this state change for each finger, which doesn't look _right_. Imagine decorations lightening, etc.

Please elaborate.
Why would it be wrong to do "decorations lightening" on the second or third finger?
Wouldn't that also mean that you shouldn't do "decorations lightening" on `PointerButton`?

Reply:
You react to `PointerEnter` not `PointerButton` and for mouse you can not get mulitple. With touch you already have decorations _lightened_ from the first touch input, the rest you should just ignore since user already interacting with the window, since you haven't got any _leave_ events.

Though, generally applications have it delivered via _separate_ event these dasy.

Reply:
> You react to `PointerEnter` not `PointerButton` and for mouse you can not get mulitple.

The point is that users should specifically **not react** to `PointerButton` in this context.
This is questionable because its not obvious to a user why they should not.

> With touch you already have decorations _lightened_ from the first touch input, the rest you should just ignore since user already interacting with the window, since you haven't got any _leave_ events.

It might help if you explain what you mean with "decorations" here.
But the logic you are talking about here seems to build upon some rules that don't seem obvious to me, I don't know why users should react to this but not to that.

Maybe this is also something we should hash out in the meeting so you can explain this whole matter holistically.